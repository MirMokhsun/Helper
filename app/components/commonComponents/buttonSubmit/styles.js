import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    wrapper: {
        height: 45,
        alignItems: 'center',
        marginBottom: 15,
        justifyContent: 'center',
        borderRadius: 24,
    },
    btnBackgroundColorGreen: {
        backgroundColor: '#1459D2',
    },
    btnBackgroundColorDarkGrey: {
        backgroundColor: '#1D2E3E',
    },
    text: {
        fontSize: 16,
        fontFamily: 'Roboto-Light',
        color: "#FFFFFF",
    },
    textColorWhite: {
        color: "#FFFFFF",
    },
    textColorGrey: {
        color: "#5A6772",
    },
    containerSpiner: {
        position: 'absolute',
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
        justifyContent: 'center',
        backgroundColor: 'rgba(0,0,0,0.5)',
        borderRadius: 24,
    },
});