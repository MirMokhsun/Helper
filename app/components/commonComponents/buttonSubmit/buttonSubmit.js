import React from 'react';
import { View, TouchableOpacity, Text, ActivityIndicator } from 'react-native';
import { styles } from './styles';

export class ButtonSubmit extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            spiner: null
        }
    }

    UNSAFE_componentWillReceiveProps({ isShowSpiner }) {
        let spiner = null;
        if (isShowSpiner) {
            spiner =
                (<View style={styles.containerSpiner} >
                    <ActivityIndicator size="large" color="#EAF1F4" />
                </View>);
        }
        this.setState({ spiner });
    }

    render() {
        const { testID, isDisabled, text, width, isShowSpiner, onPress } = this.props;
        const { spiner } = this.state;
        return (
            <TouchableOpacity disabled={(isDisabled || isShowSpiner)} onPress={onPress} testID={testID} accessibilityLabel={testID}>
                <View style={[styles.wrapper, isDisabled ? styles.btnBackgroundColorDarkGrey : styles.btnBackgroundColorGreen, { width: width || 290 }]}>
                    <Text style={styles.text}>{text}</Text>
                    {spiner}
                </View>
            </TouchableOpacity>
        )
    }
}

export default ButtonSubmit;