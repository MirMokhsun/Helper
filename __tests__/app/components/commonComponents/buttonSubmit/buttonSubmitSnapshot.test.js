import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ButtonSubmit } from '../../../../../app/components/commonComponents/buttonSubmit/buttonSubmit';

it('create snapshot ButtonSubmit', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<ButtonSubmit
        text={'Registration'}
        testID={'ButtonSubmitRegistrationID'}
        isDisabled={false}
        isShowSpiner={true}
        onPress={() => { }}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});