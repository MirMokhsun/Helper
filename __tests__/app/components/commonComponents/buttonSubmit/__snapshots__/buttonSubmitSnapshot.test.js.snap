// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot ButtonSubmit 1`] = `
<TouchableOpacity
  accessibilityLabel="ButtonSubmitRegistrationID"
  activeOpacity={0.2}
  disabled={true}
  onPress={[Function]}
  testID="ButtonSubmitRegistrationID"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "height": 45,
          "justifyContent": "center",
          "marginBottom": 15,
        },
        Object {
          "backgroundColor": "#1459D2",
        },
        Object {
          "width": 290,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto-Light",
          "fontSize": 16,
        }
      }
    >
      Registration
    </Text>
  </View>
</TouchableOpacity>
`;
